src/cairo1-compiler.ts

```diff
@@ -14,48 +25,110 @@ export const exec = (args: string) => {
     } as ProcessResult;
 };
 
-export class DockerCairo1Compiler {
-    image: Image;
-    sources: string[];
-    compilerArgs: string[];
+export class CairoCompilerDownloader {
+    private isInitialized: Promise<void>;
+    compilerDownloadPath: string;
+    compilerVersion: string;
 
-    constructor(image: Image, sources: string[], cairo1CompilerArgs?: string[]) {
-        this.image = image;
-        this.sources = sources;
-        this.compilerArgs = cairo1CompilerArgs;
+    constructor(rootPath: string, starknet: StarknetConfig) {
+        this.isInitialized = this.initialize(rootPath, starknet);
     }
 
-    protected getDockerArgs(): BindsMap {
-        const binds: BindsMap = {};
-        for (const source of this.sources) {
-            binds[source] = source;
+    async initialize(rootPath: string, starknet: StarknetConfig) {
+        this.compilerDownloadPath = starknet?.cairo1BinDir || path.join(rootPath, "cairo-compiler");
+        this.compilerVersion = starknet?.compilerVersion || (await this.getCompilerVersion());
+    }
+
+    async handleCompilerDownload(): Promise<void> {
```

Comment:
It is not clear from the method name what it is doing. It could be `ensureCompilerVersionPresent`.

Reply:
https://github.com/0xSpaceShard/starknet-hardhat-plugin/pull/402/commits/5efc50a7c425e58b74d93a39d4609ca0e4ecdc76